### PR TITLE
Authorise delegate special authority requests AB#15085

### PR DESCRIPTION
**Pull request by BronzBierd**
# Implements [AB#15085](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15085)

## Description

1. Authorise delegate requests for sepecial authority requests in medication controller.
2. Add icon to view component
3. Add functional tests for dependent timelines

## Testing

- [ ] Unit Tests Updated
- [X] Functional Tests Updated - They will fail until [AB#15109](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15109) is completed
- [ ] Not Required

### dependents:
image

### dependents/:id/timline:
With Data:
image
With no data:
image

### Functional Tests:
Dashboard:
image

Dependents timeline:
image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Testing/functional/tests/cypress/integration/e2e/timeline/dependent.js [hunk 1]
```diff
@@ -290,7 +286,8 @@ describe("Dependent Timeline Datasets", () => {
         disabledDatasetShouldNotBePresent(Dataset.Note);
         disabledDependentDatasetShouldNotBePresent(Dataset.Note);
     });
-    it("Validate (lack of) Special Authority requests on dependent timeline", () => {
+    it("Validate Special Authority requests on dependent timeline", () => {
+        enabledDatasetShouldBePresent(Dataset.LabResult);
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/e2e/timeline/dependent.js:290]: Dataset.LabResult should be Dataset.SpecialAuthorityRequest.

File: Apps/Medication/src/Startup.cs [hunk 2]
```diff
@@ -62,6 +62,7 @@ public void ConfigureServices(IServiceCollection services)
             this.startupConfig.ConfigureAuditServices(services);
             this.startupConfig.ConfigureAuthServicesForJwtBearer(services);
             this.startupConfig.ConfigureAuthorizationServices(services);
+            this.startupConfig.ConfigureDelegateAuthorizationServices(services);
```
Comment (MikeLyttle) [inline Apps/Medication/src/Startup.cs:65]: This may conflict with your other active PR.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/Medication/src/Startup.cs (hunk 2 above)
Comment (BronzBierd) [reply Apps/Medication/src/Startup.cs:65]: they may, but I'm happy to resolve the merge conflict.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/Medication/src/Startup.cs (hunk 2 above)
Comment (BronzBierd) [reply Apps/Medication/src/Startup.cs:65]: Is there any particular way you would suggest I go about handling this, it feels a little inevitable with this particular set of PRs?

Comment (MikeLyttle) [reply Apps/Medication/src/Startup.cs:65]: I'd suggest just enabling it in the first one that needs it.  The second one can rely on it being already added.

**Comment by BronzBierd** [comment]
Awaiting [AB#15109](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15109) for data to validate submission.

**Review by sslaws — APPROVED** [review]

**Review by mk08sh — COMMENTED** [review]
✅

**Pull request merged**